arlas-rest/src/main/java/io/arlas/server/rest/explore/aggregate/GeoAggregateRESTService.java

```diff
@@ -388,29 +383,39 @@ public Response geoaggregatePost(
         request.basicRequest = aggregationRequest;
         request.headerRequest = aggregationsRequestHeader;
 
-        FeatureCollection fc = getFeatureCollection(request, collectionReference, (aggregationRequest.form!=null && aggregationRequest.form.flat));
+        FeatureCollection fc = getFeatureCollection(request, collectionReference, (aggregationRequest.form != null && aggregationRequest.form.flat), Optional.absent());
 
         return cache(Response.ok(fc), maxagecache);
     }
 
-    private FeatureCollection getFeatureCollection(MixedRequest request, CollectionReference collectionReference, boolean flat) throws ArlasException, IOException {
+    private FeatureCollection getFeatureCollection(MixedRequest request, CollectionReference collectionReference, boolean flat, Optional<String> geohash) throws ArlasException, IOException {
+
+        Number precision = ((AggregationsRequest) request.basicRequest).aggregations.get(0).interval.value;
         FeatureCollection fc;
         AggregationResponse aggregationResponse = new AggregationResponse();
-        AggregationTypeEnum maintAggregationType = ((AggregationsRequest)request.basicRequest).aggregations.get(0).type;
```

Comment:
You may use an Optional to ensure that .aggregations and .aggregations.get(0) aren't null